src/ln/keysinterface.rs

```diff
@@ -0,0 +1,186 @@
+//! WalletInterface is *not* a wallet, only an interface to bridge between
```

Comment:
Hmm, can we put keysinterface in chain:: or util::? Maybe rename chain:: to something like interfaces:: and we can move logger there eventually (though thats kinda orthogonal).

Reply:
Done to chain::, I think handling onchain outputs is more L1 related than just bunch of tools like util::

Reply:
But in fact, as to move ChannelKeys in keysinterface due to scope issues?

Reply:
I mean you can move ChannelKeys into keysinterface in this PR, that doesn't feel out of scope to me, I was just saying moving logger feels like scope creep, but up to you either way.

Reply:
Done for move of ChannelKeys, will see later for logger scope!